guides/common/modules/proc_upgrading-project-or-proxy-in-place-using-leapp.adoc

```diff
@@ -0,0 +1,321 @@
+enabled=1
+gpgcheck=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
+module_hotfixes=1
+
+[leapp-foreman-client]
+name=Foreman client {ProjectVersion}
+baseurl=https://yum.theforeman.org/client/{ProjectVersion}/el8/$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman-client
+
+[leapp-puppet7]
+name=Puppet 7 Repository el 8 - $basearch
+baseurl=http://yum.puppetlabs.com/puppet7/el/8/$basearch
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-puppet7-release
+       file:///etc/pki/rpm-gpg/RPM-GPG-KEY-2025-04-06-puppet7-release
+enabled=1
+gpgcheck=1
+----
+
+* You need a Puppet repository for the Puppet agent that the installer is using.
+
+. We do not support {EL} 9 installations with EPEL 9 enabled, so remove `epel-release`:
++
+----
+# yum remove epel-release
+----
+
+. Remove `centos-release-scl` and `centos-release-scl-rh` repositories:
++
+----
+# yum remove centos-release-scl centos-release-scl-rh
+----
+endif::[]
+
+. Configure Leapp to skip updating the Tomcat packages to ensure the upgrade does not fail:
++
+----
+# echo tomcat >> /etc/leapp/transaction/to_remove
+# echo tomcat-lib >> /etc/leapp/transaction/to_remove
+----
+
+. Let Leapp analyze your system:
++
+----
+# leapp preupgrade
+----
+ifdef::satellite[]
++
+If you run {Project} in a disconnected environment, add the `--no-rhsm` and `--enablerepo` parameters:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# leapp preupgrade \
+--no-rhsm \
+--enablerepo BaseOS \
+--enablerepo AppStream \
+--enablerepo {RepoRHEL9ServerSatelliteServerProductVersion} \
+--enablerepo {RepoRHEL9ServerSatelliteMaintenanceProductVersion}
+----
+endif::[]
+
++
+The first run is expected to fail but report issues and inhibit the upgrade.
+Examine the report in the `/var/log/leapp/leapp-report.txt` file, answer all questions (using `leapp answer`), and manually resolve the other reported problems.
++
+The following commands show the most common steps required:
++
+----
+# rmmod pata_acpi
+# echo PermitRootLogin yes | tee -a /etc/ssh/sshd_config
+# leapp answer --section remove_pam_pkcs11_module_check.confirm=True
+----
++
+If `leapp preupgrade` inhibits the upgrade with *Unsupported network configuration* because there are multiple legacy named network interfaces, follow the instructions shown by Leapp to rename the interfaces, followed by an installer run to reconfigure {Project} or {SmartProxy} to use the new interface names:
++
+[options="nowrap" subs="attributes"]
+----
+# {foreman-installer} --help |grep 'interface.*eth'
+    --foreman-proxy-dhcp-interface  DHCP listen interface (current: "eth0")
+    --foreman-proxy-dns-interface  DNS interface (current: "eth0")
+----
++
+If `eth0` was renamed to `em0`, call the installer to use the new interface name with:
++
+[options="nowrap" subs="attributes"]
+----
+# {foreman-installer} --foreman-proxy-dhcp-interface=em0 --foreman-proxy-dns-interface=em0
+----
+
```

Comment:
Those are not the most common failures for el8-to-el9, so I'd kick that part and re-introduce once we get reports that anything is needed
```suggestion
```